booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -461,23 +424,44 @@ private static <T, O> Supplier<T> delayedValue(Supplier<O> cspSupplier, Function
     };
   }
 
+
   private static void generateScenarioRelatedChecks(List<JsonContentCheck> checks, Supplier<CarrierScenarioParameters> cspSupplier, Supplier<DynamicScenarioParameters> dspSupplier) {
-    checks.add(JsonAttribute.mustEqual(
+    checks.add(JsonAttribute.customValidator(
       "[Scenario] Verify that the correct 'carrierServiceName' is used",
-      "carrierServiceName",
-      delayedValue(cspSupplier, CarrierScenarioParameters::carrierServiceName)
+      (body) -> {
+        var carrierServiceName = body.path("carrierServiceName");
+        var issues = new LinkedHashSet<String>();
+        if(carrierServiceName != null) {
+          delayedValue(cspSupplier, CarrierScenarioParameters::carrierServiceName  );
+        }
+        return issues;
```

Comment:
This code does not ever issue an error for non-conformance. Accordingly, there must be a bug somewhere.

Reply:
Causing the check to never fail is not the answer. It just turns the code into dead-weight and a false sense of security.

Instead, why are you doing validation checks on a use-case where the validation check is not relevant? The scenario checks applies to all use-cases where you can check the scenario decided values, not blindly to all use-cases.

Reply:
I agree to this..